### DISCUSSION TITLE
Make sure that TCP connections to the port are not blocked by a firewall

### DISCUSSION THREAD

**Discussion by pavan-chakka** [dbeaver/dbeaver#17589] [Q&A]
### Description

Hello,

I have MAC M1 Max with Open JDK 11, and using DBeaver 22.1.2 to connect DB hosted MS SQL Server. Tried all options listed in above chat, but facing same issue and getting below error -

"The TCP/IP connection to the host XXXXXX, port 1433 has failed. Error: "XXXXXX. Verify the connection properties. Make sure that an instance of SQL Server is running on the host and accepting TCP/IP connections at the port. Make sure that TCP connections to the port are not blocked by a firewall.". "

![Screen Shot 2022-08-31 at 5 09 32 PM](https://user-images.githubusercontent.com/85208782/187803105-23df3540-a109-47d2-a2e8-82abfb9419cb.png)

Below is MS SQL Server Driver settings. Not sure what else to be done, please help if you experienced it and got any solution.

![Screen Shot 2022-08-31 at 5 28 17 PM](https://user-images.githubusercontent.com/85208782/187803151-773f509d-7d86-4e6f-9a0f-c5a70a576228.png)

![Screen Shot 2022-08-31 at 5 31 05 PM](https://user-images.githubusercontent.com/85208782/187803163-33945792-1c11-4687-8995-f8b67cfd942e.png)

### DBeaver Version

Community Edition - 22.1.2

### Operating System

Apple M1 Max - MacOs Montery - 12.5.1

### Database and driver

MS SQL Server / SQL Server
com.microsoft.sqlserver.jdbc.SQLServerDriver
com.microsoft.sqlserver:mssql-jdbc:RELEASE:9.2.0.jre8
com.microsoft.sqlserver:mssql-jdbc_auth:RELEASE:9.2.0.x64

### Steps to reproduce

1. Add SQL Database details
2. Try to connect to database by clicking 'Test Connection'
3. get this error

"The TCP/IP connection to the host XXXXXX, port 1433 has failed. Error: "XXXXXX. Verify the connection properties. Make sure that an instance of SQL Server is running on the host and accepting TCP/IP connections at the port. Make sure that TCP connections to the port are not blocked by a firewall.". "

**Comment by flambchap** [comment]
Hi. It seems like DBeaver cannot reach the specified host, or the port needs to be changed. I suggest double-checking it. Also, do you need to take some additional steps to reach the specified host (e.g., use VPN or SSH tunnel)?

**Comment by pavan-chakka** [comment]
> Hi. It seems like DBeaver cannot reach the specified host, or the port needs to be changed. I suggest double-checking it. Also, do you need to take some additional steps to reach the specified host (e.g., use VPN or SSH tunnel)?


Hello, Thanks for responding. I am on VPN only, and when I do 'ping IPADDRESS -p 1433' from terminal, I am able to establish connection and get packets. So only from DBeaver, am unable to connect.

**Comment by pavan-chakka** [comment]
I was able to connect thru IpAddress

**Comment by toniasep** [comment]
i have same problem :(

**Reply by pavan-chakka** [reply]
did you tried connecting with IP Address?

**Reply by toniasep** [reply]
iam using tunneling to connect sql server, and now problem is solved, when i add -Nv on tunneling
